src/biz/bokhorst/xprivacy/XPrivacy.java

```diff
@@ -29,64 +28,107 @@
-							if (name.length > 0) {
-								String methodName = (name.length > 1 ? name[1] : null);
-								CRestriction restriction = new CRestriction(0, name[0], methodName, null);
-								Log.w("XPrivacy", "Disabling " + restriction);
-								mListDisabled.add(restriction);
-							}
-						}
-					br.close();
-					ir.close();
-					fis.close();
-				} catch (Throwable ex) {
-					Log.w("XPrivacy", ex.toString());
-				}
-		}
-	}
-
-	// Xposed
+	private static boolean mHasErr = false;
+	
 	public void initZygote(StartupParam startupParam) throws Throwable {
 		// Check for LBE security master
 		if (Util.hasLBE()) {
 			Util.log(null, Log.ERROR, "LBE installed");
 			return;
 		}
+		
+		/*
+		 * ActivityManagerService is the beginning of the main "android" process. 
+		 * This is where the core java system is started, where the system context is created and so on. 
+		 * In pre-lollipop we can access this class directly, but in lollipop we have to visit ActivityThread first, 
+		 * since this class is now responsible for creating a class loader that can be used to access ActivityManagerService. 
+		 * It is no longer possible to do so via the normal boot class loader. Doing it like this will create a consistency between older and newer 
+		 * Android versions.
+		 * 
+		 * Note that there is no need to handle arguments in this case. And we don't need them so in case they change over time, we will simply use 
+		 * the hookAll feature. 
+		 */
+		try {
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+				XposedBridge.hookAllMethods(Class.forName("android.app.ActivityThread"), "systemMain", new XC_MethodHook() {
+					@Override
+					protected void afterHookedMethod(MethodHookParam param) throws Throwable {
+						final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+						
+						try {
+							XposedBridge.hookAllConstructors(Class.forName("com.android.server.am.ActivityManagerService", false, loader), new XC_MethodHook() {
+								@Override
+								protected void afterHookedMethod(MethodHookParam param) throws Throwable {
+									bootstrapSystem(param.thisObject, loader);
+								}
+							});
+							
+						} catch (Throwable ex) {
+							Util.bug(null, ex); mHasErr = true;
+						}
+					}
+				});
+				
+			} else {
+				XposedBridge.hookAllMethods(Class.forName("com.android.server.am.ActivityManagerService"), "startRunning", new XC_MethodHook() {
+					@Override
+					protected void afterHookedMethod(MethodHookParam param) throws Throwable {
+						bootstrapSystem(param.thisObject, null);
+					}
+				});
+			}
+			
+			bootstrapZygote();
 
-		init(startupParam.modulePath);
+		} catch (Throwable ex) {
+			Util.bug(null, ex); mHasErr = true;
+		}
 	}
-
+	
 	public void handleLoadPackage(final LoadPackageParam lpparam) throws Throwable {
 		// Check for LBE security master
-		if (Util.hasLBE())
+		int uid = Process.myUid();
+		
+		if (Util.hasLBE() || mHasErr || uid <= Process.SYSTEM_UID)
```

Comment:
Maybe I read something wrong in the original file? But it looked to me as the only thing you used "android" for, was to add a hook in AM for Lollipop? If not, then my bad :)